ebi_eva_common_pyutils/spring_properties.py

```diff
@@ -26,90 +28,132 @@ def __init__(self, maven_profile, private_settings_file):
         self.maven_profile = maven_profile
         self.private_settings_file = private_settings_file
 
+    @staticmethod
+    def _format(*key_value_maps):
+        all_params = defaultdict(list)
+        for key_value_map in key_value_maps:
+            for key in key_value_map:
+                if key_value_map[key] is not None:
+                    all_params[key.split('.')[0]].append(f'{key}={key_value_map[key]}')
+        lines = []
+        for key_type in all_params:
+            for line in all_params[key_type]:
+                lines.append(line)
+            lines.append('')
+
+        return '\n'.join(lines)
```

Comment:
I am wondering if we should also sort the lines for more predictability...

Reply:
They are already grouped based on the first part of the properties and they are otherwise in the order of insertion. 
Question is wether it's better to control the order here or have them sorted alphabetically.

Reply:
You mean, sorted when it is passed to this function? It is a judgment call but I am in favor of alphabetical sort here so that we have a predictable output format....

Reply:
Just to be clear they are already in a predictably set order. The user of this library cannot change this order. 
The only thing to discuss is what this order should be and I think alphabetical is not necessarily the best. I tend to prefer `username` then `password` together and not `password` then `port` then `username`.